cmd/mattermost/commands/server.go

```diff
@@ -34,16 +36,42 @@ func init() {
 	RootCmd.RunE = serverCmdF
 }
 
+func loadCustomDefaults() (*model.Config, error) {
+	var customDefaults *model.Config
+	customDefaultsPath := os.Getenv("MM_CUSTOMDEFAULTS")
+
+	if customDefaultsPath != "" {
+		file, err := os.Open(customDefaultsPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to open custom defaults file at %q", customDefaultsPath)
+		}
+		defer file.Close()
+
+		err = json.NewDecoder(file).Decode(&customDefaults)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to decode custom defaults configuration")
+		}
+	}
+
+	return customDefaults, nil
+}
+
 func serverCmdF(command *cobra.Command, args []string) error {
 	disableConfigWatch, _ := command.Flags().GetBool("disableconfigwatch")
 	usedPlatform, _ := command.Flags().GetBool("platform")
 
 	interruptChan := make(chan os.Signal, 1)
 
 	if err := utils.TranslationsPreInit(); err != nil {
-		return errors.Wrapf(err, "unable to load Mattermost translation files")
+		return errors.Wrap(err, "unable to load Mattermost translation files")
 	}
-	configStore, err := config.NewStore(getConfigDSN(command, config.GetEnvironment()), !disableConfigWatch)
+
+	customDefaults, err := loadCustomDefaults()
+	if err != nil {
+		return errors.Wrap(err, "cannot load custom defaults")
```

Comment:
Should we have a soft fail here, we log the error but continue the server startup? Not sure if we should stop the server because of issues with custom defaults. Thoughts @jwilander @ethervoid ?

Reply:
We started a discussion [here as well](https://github.com/mattermost/mattermost-server/pull/16265#discussion_r522272366), I'm really 0/5 on this. If we're planning to use the feature mainly on cloud and we rely on this to work, my assumption is that we would prefer for the pods not to start on error, but I'm happy to change the approach if a soft failure is preferred

Reply:
If possible we can check if the env. is cloud and make a hard fail in that case. Could be that too specific?

Reply:
I don't believe there should be anything in the defaults that are absolutely necessary for pods to start, those type of settings are handled directly by env var in the groups. @jwilander can correct me here.  

Just want to make sure we don't stop the server by an improper use of the new custom defaults env var and/or defaults file outside of cloud but at the same time would like to keep this implementation generic and not include a cloud condition here.

Reply:
I agree with @marianunez. All the critical "must be set" settings will be set by env vars so I don't think it needs to be a hard fail if we can't load the custom defaults